lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3725,8 +3725,17 @@ namespace {
 
       // It is import that we bail on an unimportable record *before* we import
       // any of its members or cache the decl.
-      if (!isCxxRecordImportable(decl))
+      if (!isCxxRecordImportable(decl)) {
+        // Anonymous unions can have valid and invalid decls inside of them.
+        // When the getters and setters for such unions are generated, if the
+        // __Unnamed_union___Anonymous_field is missing due to a field not
+        // being visited here then code in makeIndirectFieldAccessors() will
+        // assert. For now this could be considered a workaround.
+        // TODO: Investigate if there is a better way to avoid asserting.
+        if (decl->isAnonymousStructOrUnion() && decl->isUnion())
+          VisitRecordDecl(decl);
```

Comment:
I don't think this is a great solution. If there are two invalid things, we'll still visit the record, that doesn't really make sense. 

Even if we return nullptr here, we're still going to be caching the record in `VisitRecordDecl` (so that we don't get into an infinite loop). At least until #36747 lands, this will still be importing a deck that we know is unimportable. (I suspect if you delete the destructor, you'll see this still "works".)

Reply:
Yep, if the custom destructor or copy constructor in this case are deleted then it does not assert. I agree this is not a great solution, but I want to see how we can use knowledge of this sort of work around to put together good solution that can avoid the assert :-/ .